numpyro/contrib/haiku_module/__init__.py

```diff
@@ -0,0 +1,7 @@
+try:
```

Comment:
If you decide to split out `flax` and `haiku`, it is better to rename those modules to `flax` and `haiku` so we can do `from numpyro.contrib.flax import flax_module` instead of `from numpyro.contrib.flax_module import flax_module`.